src/crush/CrushWrapper.cc

```diff
@@ -1990,9 +1990,12 @@ void CrushWrapper::decode(bufferlist::iterator& blp)
 	      ::decode(weight_set->weights[l], blp);
 	  }
 	  ::decode(arg->ids_size, blp);
-	  arg->ids = (__s32 *)calloc(arg->ids_size, sizeof(__s32));
```

Comment:
Should we do the same for weight_set calloc (slightly above) and get rid of arg->weight_set_size == 0 check in get_choose_arg_weights()?  I wondered why it's there...

The kernel client checks both: https://github.com/ceph/ceph-client/blob/33e9c8dbfbcef8e4cda8e43a445e692ab7e0d8c0/net/ceph/osdmap.c#L222